modules/sievefilters/modules.php

```diff
@@ -1379,11 +1379,67 @@ public function process() {
             save_main_script($client, $main_script, $scripts);
             $client->activateScript('main_script');
             $client->close();
-            
+
             Hm_Msgs::add("Script $state");
         } catch (Exception $e) {
             Hm_Msgs::add("ERRSieve: {$e->getMessage()}");
         }
         $this->out('success', $success);
     }
 }
+class Hm_Handler_list_block_sieve_script extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_server_id'));
+        if (!$success) {
+            return;
+        }
+
+        foreach ($this->user_config->get('imap_servers') as $idx => $mailbox) {
+            if ($idx == $form['imap_server_id']) {
+                $imap_account = $mailbox;
+                break;
+            }
+        }
+
+        $email_sender = $this->request->post['sender'];
+        if (mb_strstr($email_sender, '*')) {
+            $email_sender = str_replace('*', '', $email_sender);
+        }
+
+        $factory = get_sieve_client_factory($this->config);
+        try {
+            $client = $factory->init($this->user_config, $imap_account);
+            $scripts = $client->listScripts();
+
+            if(array_search('blocked_senders', $scripts, true) === false) {
+                $client->putScript(
```

Comment:
Why do you put an empty script here when you are going to check for non-empty script later in the code?